gnucash/import-export/bi-import/dialog-bi-import.c

```diff
@@ -251,12 +251,12 @@ gnc_bi_import_fix_bis (GtkListStore * store, guint * fixed, guint * deleted,
     
     // Init control variables
     running_id = g_string_new("");
+    ignore_invoice = FALSE;
```

Comment:
I may be missing something here.
I can follow you are validating all rows in the import file. And if any one field is invalid you will skip further validation of all rows related to the invoice this particular field is part of. And a single message for that invoice is stored in info.

The rows themselves however remain in the list store.

Doesn't that mean the importer will try to import those lines anyway ?

And also a multi-line invoice may have multiple invalid values in the entry related fields. I think it would be useful if those were all reported to the user so the user can fix all the issues the import file in one go instead of having to iterate over each one as the become reported by the importer.

Reply:
If a row of an invoice does not validate, ignore_invoice is set. At the end of the processing of that row (at line 365), all rows of the same invoice are removed from the store.

I will add the reporting on all errors in an invoice (as opposed to only the first error) in a separate commit.

Reply:
I missed that. Your choice to report all errors is better.